Automattic/wp-calypso
# Launchpad: Update launchpad "Choose a domain" flow

Related to #74416

## Testing / Review Estimations

Testing: Medium
Review: Medium

## Proposed Changes

* Created a new domain upsell stepper flow to reuse the existing onboarding "Domain Selection" and "Plan selection" screens
* Modified the Launchpad "Choose a domain" task to navigate to this new domain upsell stepper flow instead of `/domains/add`

## Testing Instructions



* Checkout this branch
* Create a new launchpad-enabled site or use an existing site (write, build, or free flow) 
* Click the "Choose a domain" task
* Verify the "use your own domain link" adds the correct domain and plan to your cart on the checkout screen. Empty the cart without completing the purchase and return to the domain upsell flow through the launchpad screen. (You can use any valid domain like tinyurl.com)
* Complete the flow (select domain and any plan) and ensure the domain/plan has been purchased successfully.
* Verify the Launchpad "Choose a domain" task is marked as completed after upgrading to a paid plan
* Verify existing usage of the domains component isn't affected by these changes (`/setup/newsletter` and `/setup/link-in-bio use this component)
* Verify the "Choose a domain" task redirects to `/domains/manage` if the user is on a paid plan

**Domain selection screen**
<img width="1150" alt="CleanShot 2023-03-20 at 11 26 28@2x" src="https://user-images.githubusercontent.com/10482592/226388029-de7303b9-1dbb-40bb-9d5f-357124172ad8.png">


**Plan selection screen**
![image](https://user-images.githubusercontent.com/10482592/225671745-73346239-57d7-4c5c-a3c2-c6f2b055d9f3.png)




## Pre-merge Checklist
